devel/lazarus/Portfile

```diff
@@ -0,0 +1,131 @@
+    system -W ${worksrcpath}/components/printers \
+        "sed -i.tmp '50 s|LCLGtk2|LCLnogui\}\{\$I cupsprinters_h.inc\}\{\$ENDIF\}\{\$IFDEF LCLGtk2|g' osprinters.pas && \
+         sed -i.tmp '87 s|LCLGtk2|LCLnogui\}\{\$I cupsprinters.inc\}\{\$ENDIF\}\{\$IFDEF LCLGtk2|g'   osprinters.pas && \
+         sed -i.tmp 's|LCLGtk2|LCLnogui\}uses udlgSelectPrinter, udlgPropertiesPrinter, udlgPageSetup;\{\$I cupsprndialogs.inc\}\{\$ENDIF\}\{\$IFDEF LCLGtk2|g' ${worksrcpath}/components/printers/printersdlgs.pp "
+
+# some more fixes for nogui taking cocoa as a template
+# this may break, as soon as cocoa gets working versions
+    copy       ${worksrcpath}/components/lclextensions/include/cocoa ${worksrcpath}/components/lclextensions/include/nogui
+    reinplace "s|CocoaInt, CocoaGDIObjects, ||g"                     ${worksrcpath}/components/lclextensions/include/nogui/uses.inc
+    reinplace "s|Cocoa Interface|NoGui Interface|g"                  ${worksrcpath}/components/lclextensions/include/nogui/delphicompat.inc
+
+# Note: I did not manage a solution with reinplace
+    system -W ${worksrcpath}/components/lclextensions/include/nogui \
+        "sed -i.tmp 's|\$define HAS_GETCURRENTOBJECT|.\$define HAS_GETCURRENTOBJECT|g' delphicompat.inc && \
+         sed -i.tmp '21,27d' delphicompat.inc && \
+         sed -i.tmp '46,68d' delphicompat.inc "
+
+    copy ${worksrcpath}/components/virtualtreeview/units/cocoa         ${worksrcpath}/components/virtualtreeview/units/nogui
+    copy ${worksrcpath}/components/virtualtreeview/include/intf/cocoa/ ${worksrcpath}/components/virtualtreeview/include/intf/nogui
+
+}
+
+build.target       bigide
+build.post_args    OPT=-gl LCL_PLATFORM=cocoa
+use_parallel_build no
+
+destroot {
+    xinstall -m 755 -d ${destroot}${prefix}/etc/lazarus
+    xinstall -m 644 ${worksrcpath}/tools/install/macosx/environmentoptions.xml ${destroot}${prefix}/etc/lazarus
+
+    system -W ${worksrcpath} "make install INSTALL_PREFIX=${destroot}${prefix}"
+
+# fix the symbolic links in ${destroot}${prefix}/bin
+    system -W ${destroot}${prefix}/bin \
+        "ln -sf ../share/lazarus/lazarus      lazarus-ide  && \
+         ln -sf ../share/lazarus/startlazarus startlazarus && \
+         ln -sf ../share/lazarus/lazbuild     lazbuild     "
+
+# fix the symbolic links in the application bundle
+    system -W ${destroot}${prefix}/share/lazarus/lazarus.app/Contents/MacOS \
+        "ln -sf ${prefix}/share/lazarus/lazarus      lazarus      && \
+         ln -sf ${prefix}/share/lazarus/startlazarus startlazarus "
+
+# remove the faulty app bundle in Resources
+    delete ${destroot}${prefix}/share/lazarus/lazarus.app/Contents/Resources/startlazarus.app
+
+# copy lazarus.app to MacPorts application directory
+    copy ${destroot}${prefix}/share/lazarus/lazarus.app ${destroot}${applications_dir}
+
+# enable rebuilding from within lazarus
+    system -W ${destroot}${prefix}/share "chmod -R a+rw lazarus"
```

Comment:
What does "rebuilding from within lazarus" mean? I hope that doesn't imply that users can modify the existing files? It does seem a bit dangerous to make stuff in share world writable...

Reply:
That issue is a hard one. Against your hope, it really means that users can recompile the sources of lazarus and create new or additional .o and .ppu files. Even an accidental mismatch between .o, .ppu and source file about date or version can actually trigger the recompilation of the corresponding source files. Also, this is, how Lazarus can be reconfigured for additional packages, which change the menu layout of lazarus and extent it for other platforms (carbon, cocoa, cocoa-i386, qt4, qt5, including cross compiling for windows, wince). I tried to sort that out in a more detailed way and make the sources read-only, but did not manage this in a stable way. The reason is that folders with the object and unit files (*.o and *.ppu) are created on the fly deep in the tree of source files. IIRC, some source files are even created on demand, although few. IOW, the whole structure was too complex, to resolve reliably. I completely agree, that this is dangerous and requires discussions, but have no other practical solution without a too large effort.

Reply:
You might want to bring this issue up on the macports developer mailing list. It's sufficiently weird that it needs some discussion before we commit it as is I think.

Reply:
A corresponding email is sent to the mailing list.

Reply:
discussion is on going on mailing list. Close it here.